qualification/antenna_channelised_voltage/test_vlbi.py

```diff
@@ -0,0 +1,121 @@
+# limitations under the License.
+################################################################################
+
+"""Test filter response for VLBI narrowband mode."""
+
+import numpy as np
+import pytest
+from matplotlib.figure import Figure
+from pytest_check import check
+
+from ..cbf import CBFRemoteControl
+from ..recv import BaselineCorrelationProductsReceiver
+from ..reporter import POTLocator, Reporter
+from . import sample_tone_response_hdr
+
+
+@pytest.mark.vlbi_only
+@pytest.mark.name("VLBI filter response")
+async def test_filter_response(
+    cbf: CBFRemoteControl,
+    receive_baseline_correlation_products: BaselineCorrelationProductsReceiver,
+    pdf_report: Reporter,
+    pass_bandwidth: float,
+    pass_channels: slice,
+) -> None:
+    """Test frequency response of VLBI narrowband DDC filter.
+
+    Verification method
+    -------------------
+    Verified by means of test. A range of channels is sampled by using
+    a tone as the signal. The tones are placed in the centres of the
+    channel bins to eliminate spectral leakage from the results.
+    Check that the passband ripple is at most 0.1 dB and that no alias in the
+    passband exceeds -80 dB.
+
+    The test is performed for 1712 MHz input bandwidth, 107 MHz output
+    bandwidth and 64 MHz passband.
+    """
+    receiver = receive_baseline_correlation_products
+
+    # Limit number of samples to take - otherwise the test becomes too slow.
+    step = max(1, receiver.n_chans // 64)
+    # Sample a range covering twice the bandwidth, to determine response to
+    # aliases.
+    channels = np.arange(-receiver.n_chans // 2 + step // 2, receiver.n_chans * 3 // 2, step)
+    # Determine channel to read out response from i.e. into which the input tone
+    # will alias.
+    out_channels = channels % receiver.n_chans
+
+    pdf_report.step(f"Collect response to {len(channels)} frequencies.")
+    data = await sample_tone_response_hdr(
+        cbf=cbf,
+        receiver=receiver,
+        pdf_report=pdf_report,
+        amplitude=0.99,
+        rel_freqs=channels,
+    )
+
+    pdf_report.step("Analyse results.")
+
+    data /= np.max(data)  # Normalise
+    # data contains the complete spectral response to every input, but we're
+    # only interested in the response in the corresponding channel.
+    data = data[np.arange(len(channels)), out_channels]
+    # Make relative to centre frequency
+    freqs = receiver.channel_frequency(channels) - receiver.center_freq
+    # The maximum is to avoid errors when data is 0
+    with np.errstate(divide="ignore"):  # Avoid warnings when taking log of 0
+        data_db = 10 * np.log10(data)
+
+    # Turn pass_channels (which slices the full set of channels) into
+    # a mask over the channels collected.
+    pass_select = (pass_channels.start <= channels) & (channels < pass_channels.stop)
+    ripple = np.max(data_db[pass_select]) - np.min(data_db[pass_select])
```

Comment:
* I know you've done `data_db` to differentiate from `data`, and 
* I know `ripple` is used in the below `pdf_report.detail` with unit dB in the text

*However*, would you be open to appending an `_db` to `ripple` and `max_alias` below?
* Just so the assert has some context/info about what type of value you're checking against

`_db` appendage or a brief comment next to each `assert`, please.

Reply:
Fair point. I've appended `_db`.